### PR TITLE
[#10569] Loading icons bugs for Session Edit Form (Copy Button)

### PR DESCRIPTION
**Pull request by lyeyixian**
Part of #10569

Adds codes to change the boolean `this.sessionEditFormModel.isCopying = false` at appropriate places.
I also remove `this.failedToCopySessions = {};` as I feel it is redundant and doing nothing in the function.

**Before:**

https://user-images.githubusercontent.com/58477697/122931188-3d802080-d39f-11eb-9e54-9d925489a12f.mp4

**After:**

https://user-images.githubusercontent.com/58477697/122930901-f6922b00-d39e-11eb-970f-2beafedb4fab.mp4

**Before (error):**

https://user-images.githubusercontent.com/58477697/122931442-79b38100-d39f-11eb-8772-06c20739e340.mp4

**After (error):**

https://user-images.githubusercontent.com/58477697/122930807-e11d0100-d39e-11eb-9800-4a2182f5429c.mp4

### CODE REVIEW

File: src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts [hunk 1]
```diff
@@ -252,18 +252,19 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
...
     this.courseService.getInstructorCoursesThatAreActive()
+    .pipe(finalize(() => this.sessionEditFormModel.isCopying = false))
     .subscribe((courses: Courses) => {
       const modalRef: NgbModalRef = this.ngbModal.open(CopySessionModalComponent);
       modalRef.componentInstance.newFeedbackSessionName = this.feedbackSessionName;
       modalRef.componentInstance.courseCandidates = courses.courses;
       modalRef.componentInstance.sessionToCopyCourseId = this.courseId;
 
       modalRef.result.then((result: CopySessionModalResult) => {
-        this.failedToCopySessions = {};
```
Comment (daongochieu2810) [inline src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts:262]: Have you tried doing the following:
- Make a failed copy
- Make a successful copy
I believe this line is to remove the errors from previous actions

Comment (lyeyixian) [reply src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts:262]: oh ya, I just tested it. Yup, I will revert it back

**Review by daongochieu2810 — APPROVED** [review]
lgtm

**Comment by ccyccyccy** [comment]
The loading icon appears right after the button is clicked and disappears instantly. Could you remove that effect?

**Comment by lyeyixian** [comment]
> The loading icon appears right after the button is clicked and disappears instantly. Could you remove that effect?

Yup, I removed it.

**Review by ccyccyccy — APPROVED** [review]
LGTM!

**Comment by ccyccyccy** [comment]
@lyeyixian Can you resolve the conflict here? Thank you

**Comment by lyeyixian** [comment]
@ccyccyccy I'm not sure which to keep, both serve the same purpose
The difference is indicated with <<<<<
What I think is to use the pipe one, cos it is more consistent with the other code
```
requestList[0].subscribe((createdSession: FeedbackSession) => {
            if (Object.keys(this.failedToCopySessions).length === 0) {
              this.navigationService.navigateWithSuccessMessage(this.router,
                  '/web/instructor/sessions/edit',
                  'The feedback session has been copied. Please modify settings/questions as necessary.',
                  { courseid: createdSession.courseId, fsname: createdSession.feedbackSessionName });
            } else {
              this.statusMessageService.showErrorToast(this.getCopyErrorMessage());
              this.sessionEditFormModel.isCopying = false; <<<<
            }
          }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); });
```

or

```
 requestList[0].pipe(finalize(() => this.sessionEditFormModel.isCopying = false))<<<<<.subscribe((createdSession: FeedbackSession) => {
            if (Object.keys(this.failedToCopySessions).length === 0) {
              this.navigationService.navigateWithSuccessMessage(this.router,
                  '/web/instructor/sessions/edit',
                  'The feedback session has been copied. Please modify settings/questions as necessary.',
                  { courseid: createdSession.courseId, fsname: createdSession.feedbackSessionName });
            } else {
              this.statusMessageService.showErrorToast(this.getCopyErrorMessage());
            }
          }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); });
```

**Comment by lyeyixian** [comment]
@ccyccyccy hi I resolved the conflict, can proceed to merge.

**Pull request merged**